tools/modules/EnsEMBL/Web/Component/Tools/VR/Results.pm

```diff
@@ -0,0 +1,562 @@
+use POSIX qw(ceil);
+use EnsEMBL::Web::Utils::FormatText qw(helptip);
+
+use parent qw(EnsEMBL::Web::Component::Tools::VR);
+
+sub content {
+  my $self    = shift;
+  my $hub     = $self->hub;
+  my $sd      = $hub->species_defs;
+  my $object  = $self->object;
+  my $ticket  = $object->get_requested_ticket;
+  my $job     = $ticket ? $ticket->job->[0] : undef;
+
+  return '' if !$job || $job->status ne 'done';
+
+  my $job_data  = $job->job_data;
+  my $job_config  = $job->dispatcher_data->{'config'};
+  my $species   = $job->species;
+
+  my $ticket_name = $object->parse_url_param->{'ticket_name'};
+
+  # THIS OUTPUT IS DEFINED IN THE RUNNABLE
+  my $output_file  = 'vr_output';
+  my $output_file_json = 'vr_output.json';
+  my $output_file_vcf = 'vr_output.vcf';
+
+  my $result_headers = $job_config->{'result_headers'};
+  my @headers = @$result_headers;
+
+  # From output file in tab format
+  my @content = file_get_contents(join('/', $job->job_dir, $output_file), sub { s/\R/\r\n/r });
+
+  my %params = map { $_ eq 'update_panel' ? () : ($_ => $hub->param($_)) } $hub->param;
+  my $html = '';
+
+  my @rows = ();
+  foreach my $line (@content) {
+    chomp $line;
+    next unless $line !~ /^#/;
+    my @split     = split /\t/, $line;
+    my %row_data  = map { $headers[$_] => $split[$_] } 0..$#headers;
+    push @rows, \%row_data;
+  }
+
+  # niceify for table
+  my %header_titles = (
+    'id'                  => 'Variant identifier',
+    'hgvsg'               => 'HGVS Genomic',
+    'hgvsc'               => 'HGVS Transcript',
+    'hgvsp'               => 'HGVS Protein',
+    'vcf_string'          => 'VCF format',
+    'spdi'                => 'SPDI',
+    'allele'              => 'Allele',
+    'input'               => 'Uploaded variant'
+  );
+  for (grep {/\_/} @headers) {
+    $header_titles{$_} ||= $_ =~ s/\_/ /gr;
+  }
+
+  my %field_descriptions = (
+    'id'                  => 'dbSNP rsIDs or any synonym for a variant present in the Ensembl Variation database',
+    'hgvsg'               => 'HGVS genomic sequence name',
+    'hgvsc'               => 'HGVS coding sequence name',
+    'hgvsp'               => 'HGVS protein sequence name',
+    'vcf_string'          => 'Position based name',
+    'spdi'                => 'Genomic SPDI notation: NCBI variation notation described as Sequence Position Deletion Insertion (https://www.ncbi.nlm.nih.gov/variation/notation/)',
+    'allele'              => 'Alternative allele',
+    'input'               => 'Uploaded variant identifier'
+  );
+
+  my @table_headers = map {{
+    'key' => $_,
+    'title' => ($header_titles{$_} || $_),
+    'sort' => 'string',
+    'help' => $field_descriptions{$_}
+  }} @headers;
+
+  my $new_rows;
+
+  $params{'line_count'} = 0 unless defined ($params{'parsed'});
+
+  my $output_lines = scalar(@rows);
+
+  my ($new_rows, $line_count) = $self->get_rows(\%params, \@rows, $output_lines);
+
+  my @rows_2 = @$new_rows;
```

Comment:
I would suggest to improve the variable name of $new_rows. I also don't understand the purpose of the method get_rows. I'm happy to have a chat about this.
I would advice against using @rows_2 as a variable name.